.github/actions/cluster-init/action.yaml

```diff
@@ -0,0 +1,102 @@
+name: Cluster Init
+description: Set up Turing Cluster Init
+
+inputs:
+  cluster_name: 
+    required: true
+    description: 'Name of Cluster'
+    default: ''
+  istio_version:
+    required: true
+    description: 'Istio Version'
+    default: ''
+  knative_version:
+    required: true
+    description: 'Knative Version'
+    default: ''
+  knative_istio_version:
+    required: true
+    description: 'Knative Istio Version'
+    default: ''
+  local_registry:
+    required: true
+    description: 'Endpoint of local registry'
+    default: ''
+  build_image:
+    required: false
+    description: '(Optional) Flag to build image from code'
+    default: false
+  cluster_init_version:
+    required: false
+    description: '(Optional) Required if build_image is false, Version of cluster to install  '
+    default: ''
+
+  
+runs:
+  using: composite
+  steps:
+    - name: "Setup local k8s cluster"
+      uses: AbsaOSS/k3d-action@v1.5.0
+      with:
+        cluster-name: ${{ inputs.cluster_name }}
+        use-default-registry: true
+        args: >-
+          --servers 1
+          --agents 3
+          --port 80:80@loadbalancer
+          --k3s-server-arg "--no-deploy=traefik,metrics-server"
+
+    - name: Build Docker image
+      if: ${{ inputs.build_image == 'true' }}
```

Comment:
Need to do a string check, as it is actually a string and not a boolean - https://github.com/actions/runner/issues/1483